### PR TITLE
Extend Apartment model

### LINKED ISSUE #184: As user, I want update the Zillow model to be a generic Apartment model so that we can search against them and have a common model too include all externally-sourced data.
**User story**

As user, I want update the Zillow model to be a generic Apartment model so that we can search against them and have a common model too include all externally-sourced data.

**Acceptance criteria**

**Definition of Done**

I know this is done when the ZillowHousing model has been updated to the Generic "Apartments" type. This will include all fields from zillow as well as any fields we pull in from other sources like craigslist.

### PR DESCRIPTION
**Pull request by alldne**
Closes #184 

This PR changes Apartment model:
- adds `number_of_bed`, `rent_price` field
- deletes `estimated_rent_price_currency` to make it simple since we're only targeting the U.S.
- makes all the zillow-related field optional (null=True)

Also this PR:
- removes zillow informations from Apartment model when it's out of date
- creates or updates Apartment model by suite_num instead of zpid because zpid is now optional

### CODE REVIEW

**Comment by ab7289** [comment]
Hmm, I added an apartment to `/location/1` in the admin panel but when I navigated to /location/1` in the site it didn't pop up in the list. Are we not pulling apartments from the database yet?

**Comment by alldne** [comment]
@ab7289 Currently the site will show only apartments with zillow information (technically it checks only if it has `zpid` tho)

That's because in our location page, zillow link is mandatory and I've not changed the page itself in this PR.

File: external/cache/tests.py [hunk 1]
```diff
@@ -60,11 +66,50 @@ def test_cache(self, actual_request):
...
+        )
+
+        # 1. # apt with zpid = 1
+        self.assertEqual(loc.apartment_set.exclude(zpid=None).count(), 1)
+
+        # 2. refresh_zillow_housing_if_needed should
+        # - add zillow information in the response (+result_count)
+        # - reset the outdated zillow information (-1)
+        refresh_zillow_housing_if_needed(loc)
+
```
Comment (KShih) [inline external/cache/tests.py]: 👍

File: location/views.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def get_object(self):
     def get_context_data(self, **kwargs):
         context = super(LocationView, self).get_context_data(**kwargs)
         context["form"] = ReviewForm()
+        context["zillow_list"] = self.object.apartment_set.exclude(zpid=None).all()
```
Comment (KShih) [inline location/views.py]: 👍

File: location/models.py [hunk 3]
```diff
@@ -31,22 +32,29 @@ def google_map_url(self):
 
 
 class Apartment(models.Model):
-    suite_num = models.CharField(max_length=30, blank=True, null=True)
+    suite_num = models.CharField(unique=True, max_length=30, blank=True, null=True)
```
Comment (KShih) [inline location/models.py]: Just wondering what is the reason to set the `unique=True` here

Comment (alldne) [reply location/models.py]: It's because I assumed there is no duplicate apartment with the same suite num.

**Pull request merged**